casbin/enforcer.cpp

```diff
@@ -206,8 +218,31 @@ Enforcer::Enforcer(const std::shared_ptr<Model>& m, std::shared_ptr<Adapter> ada
 
     this->Initialize();
 
-    if (m_adapter && m_adapter->file_path != "")
-        this->LoadPolicy();
+    // load policy if needed
+    if (m_adapter)
+    {
+        // try understand if we should load policy now, this depends on a type of the adapter
+        bool loadPolicyNow = false;
+
+        /* It was better to add function ShouldLoadPolicyNow() for all adapters, 
+         * but it is too big change for this task.
+         */
+
+        // verify if it is stringstream adapter
+        auto downcastedPtr = std::dynamic_pointer_cast<StringstreamAdapter>(m_adapter);
+        if(downcastedPtr)
```

Comment:
I guess you don't have to downcast, simply check if `m_policy_stringstream` is set or not.
Will that be enough?

Reply:
this downcast is to be sure which check for sure is in charge:
if it is StringstreamAdapter, m_policy_stringstream should not be empty.
for all other cases it is filepath.

logically, this downcasting is necessary, but the questiion will we use some day filepath or not.